### PR TITLE
ppp: 2.4.8 -> 2.4.9

### PR DESCRIPTION
**Pull request by mweinelt**
###### Motivation for this change

https://github.com/ppp-project/ppp/releases/tag/2.4.9

lcp-echo-adaptive & replacedefaultroute :partying_face: 

Built on glibc and musl.

```
What's new in ppp-2.4.9.
************************

* Support for new EAP (Extensible Authentication Protocol) methods:
  - Support for EAP-TLS, from Jan Just Keijser and others
  - Support for EAP-MSCHAPv2, from Eivind Næss, Thomas Omerzu, Tijs
    Van Buggenhout and others

* New pppd options:
  - chap-timeout
  - chapms-strip-domain
  - replacedefaultroute
  - noreplacedefaultroute
  - ipv6cp-accept-remote
  - lcp-echo-adaptive
  - ip-up-script
  - ip-down-script
  - ca
  - capath
  - cert
  - key
  - crl-dir
  - crl
  - max-tls-version
  - need-peer-eap

* Fixes for CVE-2020-8597 and CVE-2015-3310.

* libpcap is now required when compiling on Linux (previously, if
  libpcap was not present, pppd would be compiled without packet
  filtering support).

* The rp-pppoe plugin has been renamed to pppoe, to distinguish it
  from the upstream rp-pppoe code.  Its options have changed names,
  but the old names are kept as aliases.

* The configure script now supports cross-compilation.

* Many bug fixes and cleanups.
```

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by mweinelt** [comment]
Experiencing a lack of IPv6 connectivity and the systemd unit isn't happy, neither am I.

```
Jul 20 21:58:12 io pppd[8060]: rcvd [IPV6CP ConfAck id=0x1 <addr fe80::e8d5:24b4:fa14:db27>]
Jul 20 21:58:12 io pppd[8060]: sif6addr: setting remote peer address failed: Address family not supported by protocol
Jul 20 21:58:12 io pppd[8060]: sif6addr failed
Jul 20 21:58:12 io pppd[8060]: sent [IPV6CP TermReq id=0x2 "Interface configuration failed"]
Jul 20 21:58:12 io pppd[8060]: rcvd [IPV6CP TermAck id=0x2]
```

File: pkgs/tools/networking/ppp/default.nix [hunk 1]
```diff
@@ -1,42 +1,27 @@
 { lib, stdenv, fetchurl, fetchpatch, fetchFromGitHub, substituteAll, libpcap, openssl }:
```
Comment (SuperSandro2000) [inline pkgs/tools/networking/ppp/default.nix:1]:
```suggestion
{ lib, stdenv, fetchFromGitHub, substituteAll, libpcap, openssl }:
```

File: pkgs/tools/networking/ppp/default.nix [hunk 2]
```diff
@@ -1,42 +1,27 @@
...
   src = fetchFromGitHub {
     owner = "paulusmack";
     repo = "ppp";
     rev = "ppp-${version}";
-    sha256 = "1i88m79h6g3fzsb4yw3k8bq1grsx3hsyawm7id2vcaab0gfqzjjv";
+    sha256 = "sha256-8+nbqRNfKPLDx+wmuKSkv+BSeG72hKJI4dNqypqeEK4=";
   };
 
   patches =
     [
```
Comment (SuperSandro2000) [inline pkgs/tools/networking/ppp/default.nix:15]:
```suggestion
  patches = [
```

Comment (mweinelt) [reply pkgs/tools/networking/ppp/default.nix:15]: When will you fix `nixos-generate-config`, which does exactly the same in hardware-configuration.nix?

**Comment by mweinelt** [comment]
IPv6 configuration now happens via rtnetlink, so we just needed to add AF_NETLINK into the mix.

**Sent over the internet line using pppd-2.4.9**

**Review by fooker — APPROVED** [review]
Works flawlessly.

**Pull request merged**